### ISSUE TITLE
Allow to have playwright tests

### ISSUE THREAD

**Issue by kaystrobach** [louislam/uptime-kuma#4804]
### 📑 I have found these related issues/pull requests

None

### 🏷️ Feature Request Type

New monitor

### 🔖 Feature description

As a user i want to run playwright tests with kuma in order to check more complex szenarios.

this is also available in other monitoring solutions:

* https://x.com/shakalandy/status/1701112612170223709?s=46&t=U8zjU-fWbJNNX4kMhADeBA

### ✔️ Solution

In browser playwright tests. It’s ok, to copy them in a textarea, the best solution would be to connect a git repo to get the tests.

### ❓ Alternatives

None, without switching products or having a middleware to run tests.

**Comment by CommanderStorm** [comment]
I think what you are asking about (especially given the asked git integration) is likely better served by a CI-piepeline

**Comment by kaystrobach** [comment]
Agree. Git is completely optional. Main use case is very simple tests, like login to idm, open service. 

The ci pipeline runs anyway. But can not run every minute etc.

other uptime checkers start to integrate these simple test case implementations to have better insights.

**Comment by kaystrobach** [comment]
related #3056

**Comment by kaystrobach** [comment]
If it helps i can create a short video showing the use case. Please just tell me. Thanks

**Comment by kaystrobach** [comment]
Other similar tools implementing playwright:

* https://betterstack.com/docs/uptime/playwright-monitor/
* https://www.checklyhq.com/product/start-monitoring-with-playwright/

**Comment by techmattr** [comment]
I too would love to see more complex checks available. I mentioned this on Reddit and was advised it was being tracked here. Currently Uptime Kuma has very limited use cases for me as all it can do is make sure my login pages are available. If Uptime Kuma had a way to log into these sites it could validate the site is actually working. The [playwright integration changedetection.io](https://i.imgur.com/IsmK1hJ.png) is done amazingly well and a similar feature would add so much value to Uptime Kuma.

**Comment by kaystrobach** [comment]
@louislam I have a developer set on that topic, if that is interesting for you, he can build the whole PoC and finalize it in the next weeks. Would be a thank you from the https://github.com/fucodo/ and https://github.com/4viewture/ team

**Comment by pilulu** [comment]
Hi,

I'm really interested in a feature like this in order to monitor small things that need a javascript.
I agree that a complex process like logging or loading multiple pages is more adapted on a CI process.
I my use case i would only be sure that all elements on a page that need javascript rendering are loaded properly.

**Comment by kaystrobach** [comment]
@louislam once I have the commitment, that this would get integrated, if implemented well, I would set a dev on that task in the next weeks, maybe for v2+?

**Comment by eitchugo** [comment]
Hi folks! I'm also interested in this, using the uptime-kuma UI to handle a user journey test on a site is much better and user-friendly than using on CI and having to setup cronjobs, etc.

Is there any plans to continue this issue? Or maybe you need any more help?

**Comment by morla10111** [comment]
Me too would love to see this feature implemented.

**Comment by mabed-fr** [comment]
I find this request interesting for creating robust tests.

In my case, it would be, for example, to add a product to the cart and open the cart. At first, we could limit the scenarios to 30 seconds.

What is the opinion of the managers and contributors on this feature, which is appearing in many monitoring software?

Considering that today it is very easy to record directly from the browser, we would not need an Iptime Kuma window.

**Comment by kaystrobach** [comment]
totally agree, should be used for short stories only. otherwise we can use external hooks now. with the new hooks, we are using gitlab ci to keep the state up to date

**Comment by sule26** [comment]
+1

**Comment by olivernz** [comment]
Did this end up going anywhere? Was just about to raise a feature request.
